appengine/pusher-chat/README.md

```diff
@@ -0,0 +1,49 @@
+# Pusher sample for Google App Engine
+
+This sample demonstrates how to use the [Pusher][pusher] on [Google App Engine][ae-docs].
+Pusher enables you to create public / private channels with presence information for real time messaging.
+This application demonstrates presence channels in Pusher using chat rooms.
+All users joining the chat room are authenticated using the `/authorize` endpoint.
+All users currently in the chat room receive updates of users joining / leaving the room.
+[Java HTTP library](https://github.com/pusher/pusher-http-java) is used for publishing messages to the channel
+and the [JS Websocket library](https://github.com/pusher/pusher-js) is used for subscribing.
+
+[pusher]: https://pusher.com
+[ae-docs]: https://cloud.google.com/appengine/docs/java/
+
+## Setup
+
+Install the [Google Cloud SDK](https://cloud.google.com/sdk/) and run:
+```
+   gcloud init
+```
+If this is your first time creating an App engine application:
+```
+   gcloud app create
+```
+
+#### Setup Pusher
+
+- Create a [Pusher] application and note down the `app_id`, `app_key`, `app_secret` and the cluster.
+- Update [appengine-web.xml](src/main/webapp/WEB-INF/appengine-web.xml) with these credentials.
+- Update Line 71 in [chat.jsp](src/main/webapp/WEB-INF/view/chat.jsp) with the `app_id` and cluster information.
```

Comment:
I wanted to keep the client side initialization simple and decoupled from the server-side to keep it easier to understand. It would be possible to do it via `ServletContextListener`  and then access the data in both places. I can see it being a problem for testing though, let me know what you think.